pvanalytics/quality/irradiance.py

```diff
@@ -471,3 +471,133 @@ def daily_insolation_limits(irrad, clearsky, daily_min=0.4, daily_max=1.25):
         lower_bound=daily_min
     )
     return good_days.reindex(irrad.index, method='pad', fill_value=False)
+
+
+def _upper_poa_global_limit_lorenz(aoi, solar_zenith, dni_extra):
+    r"""Function to calculate the upper limit of GTI
+    """
+    # Changing aoi to 90 degrees when solar zenith is greater than 90 (sun
+    # below horizon) or aoi is greater than 90 (which is not realistic).
+    aoi = aoi.mask(solar_zenith > 90, 90)
+    aoi = aoi.clip(lower=0, upper=90)
+
+    # Determining the upper limit
+    upper_limit = 0.9 * dni_extra * (cosd(aoi))**1.2 + 300
+
+    # Setting upper limit as 0 when solar zenith is > 90 (night time)
+    upper_limit[solar_zenith > 90] = 0
+
+    # Setting upper limit as undefined where solar_zenith is not available
+    upper_limit[solar_zenith.isna()] = np.nan
+
+    # Renaming upper_limit series to 'upper_limit'
+    upper_limit.rename('upper_limit')
+    return (upper_limit)
```

Comment:
```suggestion
    return upper_limit
```
Just a style nitpick, and same for the other functions